lib/vulndbhq/response/parse_json.rb

```diff
@@ -17,7 +17,10 @@ def parse(body)
 
       def on_complete(env)
         if respond_to?(:parse)
-          env[:body] = parse(env[:body]) unless [204, 304].include?(env[:status])
+	    # only call the parse routine if the response is json
+            if env[:response_headers].has_key?("content-type") && env[:response_headers]["content-type"].include?("json")
```

Comment:
Some messages are not json, so filter only messages that are json.  Additionally, added 404 to blacklist